packages/property-grid/src/api/PropertiesGridSelectionContext.ts

```diff
@@ -0,0 +1,133 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
+*--------------------------------------------------------------------------------------------*/
+import { WidgetState } from "@bentley/ui-abstract";
+import { BeEvent, Id64 } from "@bentley/bentleyjs-core";
+import {
+  FrontstageDef,
+  WidgetDef,
+  FrontstageManager,
+} from "@bentley/ui-framework";
+import { FindSimilar } from "@bentley/itwin-find-similar";
+
+/** Used to control properties grid automatic opening based on selection set changes */
+export class PropertiesGridSelectionContext {
+  private static _priorityWidgetIds: string[];
+  private static _propertyPanelId: string;
+  private static _propertyWidgetLastState: WidgetState = WidgetState.Open;
+  private static _selectionEvent:
+    | BeEvent<(ids: Set<string>) => void>
+    | undefined;
+  private static _suspended: boolean = false;
+
+  /**
+   * Initialize the properties grid selection context to provide automatic opening and closing of the property panel
+   * @param _selectionEvent Event to use for selection set change listening
+   * @param __propertyPanelId App given Id for the property panel widget
+   * @param widgetIds Ids of widgets that should take precendence and avoid the panel from opening at all if open (e.g. Find Similar Widget)
```

Comment:
update comment here depending on what we do with find-similar

Reply:
decided to go ahead and remove find similar and allow users to add their own context menu options (therefore i adjusted this comment)